### PR TITLE
Add YAML configuration options

### LINKED ISSUE #30: YAML configuration option
It would be nice if one could statically configure the daemon via YAML (or some other format, I guess). That improves the configuration management story, and also makes it a bit easier to understand what the daemon is doing without opening and reading a directory full of ruby files.

### PR DESCRIPTION
**Pull request by ahayworth**
This set of changes allows litmus_paper to be configured via ruby
DSL, YAML files, or a mixtures of both (but obviously, not in the same
file).

We only load a file as yaml if its extension is .yml or .yaml; otherwise
we evaluate it as a ruby file.

Fixes #30

### CODE REVIEW

File: lib/litmus_paper/configuration_file.rb [hunk 1]
```diff
@@ -45,12 +51,82 @@ def service(name, &block)
       @services[name.to_s] = service
     end
 
+    def yaml_service(value)
+      if value.class == Array
```
Comment (lollipopman) [inline lib/litmus_paper/configuration_file.rb]: Why is an Array of services supported in addition to a Hash?

Comment (ahayworth) [reply lib/litmus_paper/configuration_file.rb]: ...I think that was actually leftover from some iteration. I'll remove that. Good catch.

File: lib/litmus_paper/configuration_file.rb [hunk 2]
```diff
@@ -45,12 +51,78 @@ def service(name, &block)
...
     end
+
+    def load_yaml(contents)
+      config = Util.symbolize_keys(YAML.load(contents))
+
+      config.each do |key, value|
+        case key
+        when :include_files, :data_directory, :cache_location, :cache_ttl, :port
+          send(key, value)
+        when :services, :service
```
Comment (lollipopman) [inline lib/litmus_paper/configuration_file.rb]: Given that you can put multiple services in a `service` hash I think only supporting the `services` config word is less confusing.

File: lib/litmus_paper/configuration_file.rb [hunk 3]
```diff
@@ -45,12 +51,78 @@ def service(name, &block)
       @services[name.to_s] = service
     end
 
+    def yaml_service(value)
+      value.each do |name, config|
+        config = Util.symbolize_keys(config)
+        checks = []
+        dependencies = config.fetch(:dependencies, []).map do |dep|
```
Comment (lollipopman) [inline lib/litmus_paper/configuration_file.rb]: would breaking up the dependencies & checks evaluations into separate methods make the `yaml_service` method easier to read?

File: lib/litmus_paper/configuration_file.rb (hunk 2 above)
Comment (ahayworth) [reply lib/litmus_paper/configuration_file.rb]: Agreed.

File: lib/litmus_paper/configuration_file.rb (hunk 3 above)
Comment (ahayworth) [reply lib/litmus_paper/configuration_file.rb]: Slightly, and I've split them.

**Pull request merged**